js/core.js

```diff
@@ -857,14 +857,54 @@
             return this._shortestKey;
         },
 
+        /**
+         * The columns that create the shortest key that can be used for display purposes.
+         *
+         * @type {ERMrest.Column[]}
+         */
+        get displayKey () {
+            if (this._displayKey === undefined) {
+                if (this.keys.length() !== 0) {
+                    var countTextColumns = function(key) {
+                        for (var i = 0, res = 0; i < key.colset.columns.length; i++) {
```

Comment:
Didn't know you could instantiate multiple variables here. Good way to communicate the scope of a variable that isn't just the looping variable.